nullaway/src/main/java/com/uber/nullaway/handlers/contract/ContractHandler.java

```diff
@@ -107,110 +171,194 @@ public NullnessHint onDataflowVisitMethodInvocation(
     Preconditions.checkNotNull(analysis);
     Symbol.MethodSymbol callee = ASTHelpers.getSymbol(node.getTree());
     Preconditions.checkNotNull(callee);
-    // Check to see if this method has an @Contract annotation
-    String contractString = ContractUtils.getContractString(callee, config);
-    if (contractString != null && contractString.trim().length() > 0) {
-      // Found a contract, lets parse it.
-      String[] clauses = contractString.split(";");
-      for (String clause : clauses) {
-
-        MethodInvocationTree tree = castToNonNull(node.getTree());
-        String[] antecedent =
-            getAntecedent(clause, tree, analysis, state, callee, node.getArguments().size());
-        String consequent = getConsequent(clause, tree, analysis, state, callee);
-
-        // Find a single value constraint that is not already known. If more than one arguments with
-        // unknown
-        // nullness affect the method's result, then ignore this clause.
-        int argIdx = -1;
-        Nullness argAntecedentNullness = null;
-        boolean supported =
-            true; // Set to false if the rule is detected to be one we don't yet support
-
-        for (int i = 0; i < antecedent.length; ++i) {
-          String valueConstraint = antecedent[i].trim();
-          if (valueConstraint.equals("_")) {
-            continue;
-          } else if (valueConstraint.equals("false") || valueConstraint.equals("true")) {
+    MethodInvocationTree tree = castToNonNull(node.getTree());
+    for (String clause : ContractUtils.getContractClauses(callee, config)) {
+
+      String[] antecedent =
+          getAntecedent(clause, tree, analysis, state, callee, node.getArguments().size());
+      String consequent = getConsequent(clause, tree, analysis, state, callee);
+
+      // Find a single value constraint that is not already known. If more than one arguments with
+      // unknown
+      // nullness affect the method's result, then ignore this clause.
+      Node arg = null;
+      Nullness argAntecedentNullness = null;
+      boolean supported =
+          true; // Set to false if the rule is detected to be one we don't yet support
+
+      for (int i = 0; i < antecedent.length; ++i) {
+        String valueConstraint = antecedent[i].trim();
+        if (valueConstraint.equals("_")) {
+          continue;
+        } else if (valueConstraint.equals("false") || valueConstraint.equals("true")) {
+          if (arg != null) {
```

Comment:
I understand what is going on here with `false` and `true` antecedents at a high level.  Question: have you seen cases in your code base(s) where this kind of targeted handling of null checks passed to boolean parameters would be useful in terms of improving NullAway's inference / dataflow analysis?